### ISSUE TITLE
Nested notebooks displays all notes recursively in Note list

### ISSUE THREAD

**Issue by KimMichaelJones** [HappenApps/Quiver#989]
Selecting a notebook in the Sidebar (left pane) filters the list of notes in the note list (middle pane). With nested notebooks this includes all notes within all children notebooks. It would be preferable if this only displayed the notes in that notebook.

**Comment by rexwangcc** [comment]
Strongly agree with this!!! With `Nested notebooks displays all notes recursively in Note list`, my nested notebooks now look like a mess. Could you add this feature, at least as an option? This could be a great improvement! @ylian

**Comment by ylian** [comment]
I might have overthought this feature. Does anyone *prefer* the current behavior (display all notes recursively in a notebook)? I am trying to decide whether I should change the behavior or make it a setting.

**Comment by nschneid** [comment]
If it's an option, I think a natural place in the UI would be as a toggleable display option for each notebook, along with sorting. "Show nested notes" or similar. But I'm not sure how often I'd use it.

**Comment by kevanchristmas** [comment]
Assuming it doesn't affect search at all I can't think of a use case where I want to see all notes for a higher level notebook - so +1 for change behavior and don't bother adding unnecessary options/features/settings.

**Comment by ylian** [comment]
Let me make sure I get this right. 

So the real complaint here is seeing the recursive notes in the note list. For all other functions such as search inside a notebook, showing the note count of a notebook, the current recursive behavior is desired.

Please correct me if I misunderstood.

**Comment by kevanchristmas** [comment]
@ylian - for me personally you are correct, and that is also my understanding of @KimMichaelJones original comment

**Comment by rexwangcc** [comment]
I know it may be hard to do this, since the leftmost column is showing notebooks, and the second column from the left is showing notes only. If we'd like to prevent those top-level notebooks from recursively searching and showing all notes in their child-notebooks, as far as I concerned, there could be two possible solutions:
1. Enable the second column from the left to show not only notes but also notebooks, whereas this may end up with a confusing view of the second column. Below is a simple example:
```
Top-level notebook/
note 1
-- 2nd level notebook/
-- note 2
----3rd level notebook/
------ note 3
```
2. Prohibit users from having notes in top-level notebooks, this will be much clearer, but I assume this would require more work from developers, and could possibly bring more issues:
```
Top-level notebook/
[NO NOTEs ALLOWED HERE]
-- 2nd level notebook/
[NO NOTEs ALLOWED HERE]
----3rd level notebook/
------ note 1
------ note 2
```

Besides, @ylian you are correct for me personally, all other functions and the idea of having nested notebooks are really awesome.

**Comment by KimMichaelJones** [comment]
My preference would be to not add the option as I believe in keeping the UI simple and intuitive (less is more). It appears my original comment wasn't completely clear, here is a simple example that might help:

<img width="495" alt="screen shot 2017-11-14 at 11 33 38 am" src="https://user-images.githubusercontent.com/9641621/32797695-cbe1fcaa-c92f-11e7-9539-daa1f6d25388.png">
<img width="498" alt="screen shot 2017-11-14 at 11 33 29 am" src="https://user-images.githubusercontent.com/9641621/32797696-cbf37f7a-c92f-11e7-8bd8-bfd851ba7807.png">
<img width="496" alt="screen shot 2017-11-14 at 11 33 17 am" src="https://user-images.githubusercontent.com/9641621/32797697-cc04479c-c92f-11e7-8ca1-1f8cb2685b1e.png">

What I would expect / prefer is that selecting 'notebook 1' in the Sidebar (left pane) that only note 'note 1' would be displayed in the note list (middle pane). I would also expect that the counts would let me know both the note in the notebook and all descendants, eg. 1/6 for notebook 1 in this example.

**Comment by JMichaelTX** [comment]
> What I would expect / prefer is that selecting 'notebook 1' in the Sidebar (left pane) that only note 'note 1' would be displayed in the note list (middle pane). I would also expect that the counts would let me know both the note in the notebook and all descendants, eg. 1/6 for notebook 1 in this example.

I agree.  

Showing all notes in all sub-notebooks is confusing.

However, when doing a search in a notebook, we need the option to include/exclude sub-notebooks in that search.

**Comment by rexwangcc** [comment]
> What I would expect/prefer is that selecting 'notebook 1' in the Sidebar (left pane) that only note 'note 1' would be displayed in the note list (middle panel). I would also expect that the counts would let me know both the note in the notebook and all descendants, eg. 1/6 for notebook 1 in this example.

This should be an elegant solution.

**Comment by ylian** [comment]
Thanks to everyone for the comments.

I think we have consensus (me included) on the following:
1. The current behaviour of showing all notes of sub-notebooks in the note list is confusing. Esp. there is no way to see which notes are actually inside this notebook.
2. When searching inside a notebook, it's usually preferred to search through all sub-notebooks as well.

In addition, I think the basic design of the three-column layout should be kept: the leftmost column (sidebar) shows only notebooks, the middle column (note list) shows only notes. This design keeps the UI easy to understand.

I think the fundamental issue to fix here is: "showing all notes from sub-notebooks in the note list is confusing".

One way to fix this, as everyone suggested, is to display only the notes in the selected notebook, not its sub-notebooks. But this solution has a few issues:

1. If a parent notebook has no notes of its own, selecting it in the sidebar will display "No notes" in the middle column. This is confusing to me.
2. There is a mismatch between the displayed note count, and the number of notes in the note list. Admittedly a notion like "1/6", as @KimMichaelJones suggested, can solve this issue.
3. Searching inside a notebook is also confusing. Am I searching through only the notes in this notebook or all its sub-notebooks as well? More UI controls would be needed to clarify this.

An alternative solution, I suggest, is to keep displaying all notes from sub-notebooks in the note list, but group them into sections so it's not confusing.

Something like this,
<img width="491" alt="screen shot 2017-11-14 at 10 17 49 pm" src="https://user-images.githubusercontent.com/984895/32805426-aed6131c-c989-11e7-9455-bab97ed01b5d.png">

**Comment by KimMichaelJones** [comment]
Great solution @ylian, definitely works for me!

**Comment by rexwangcc** [comment]
@ylian This solution is just what I want! More ideally, it would be great if the sections, such as `WEB DEVELOPMENT`, `REACT`, `WEB CLIPPER` in your screenshot could be clickable to collapse (maybe by default they are collapsed) and expand.

**Comment by ylian** [comment]
@rexwangcc I agree that sections should be collapsible/expandable by clicking on the section header.

**Comment by matchavez** [comment]
@ylian I like the solution generally, but I have an opinion as to the "notebook". If you're going to have multiple levels of the file/folder paradigm, you either need to (IMO):

1) Designate the top level (e.g. "Web Development") as a group/bundle of notebooks, but not a place to store notes
*-or-*
2) If the top level can accept notes, ensure that the second-column list clearly indicates that the notebooks in the second-level are "below" the top notebook (maybe the twist-triangle suggested by @rexwangcc is sufficient?) and that when you click on a second-level note, the active notebook indicator moves to highlight the appropriate notebook when selected. (e.g. clicking "$.ajax" would change the indicated notebook from "Web Development" to "Web Clipper")

I suggest this, preferring the second, because this will scale to more than a two-level scenario. You could have many levels of notebooks, though 2 (maybe 3) is probably optimal.

Finally, I think it's important for count that in this example, it should be (2/65) as @KimMichaelJones  says.

**Comment by JMichaelTX** [comment]
> I agree that sections should be collapsible/expandable by clicking on the section header.

@ylian , OK that's good.  Then the behavior of NBs and sub-NBs would be the same as the Finder's Folders and sub-folders.

The subfolders definitely need to be collapsed by default, otherwise you could have an overwhelming long list of Notes.

**Comment by KimMichaelJones** [comment]
@ylian, not certain how others use the nested notebooks but have to say I appreciate and use the N-level structure with notes at each level, including the top level notebook.

**Comment by ylian** [comment]
@KimMichaelJones That is expected. I personally don't place any notes in top-level notebooks, but since there is no way to prohibit users from adding notes to a top-level notebook (and you could also create such situation simply by moving a notebook into another non-empty notebook), this use case surely needs to be supported.

I think in general making notebooks/sub-notebooks similar to Finder's folders/sub-folders is intended. It's a familiar paradigm for most users.

**Comment by meflouu** [comment]
So what was done about this in 8 months?

**Comment by meflouu** [comment]
So what was done about this?

**Comment by gbritting** [comment]
Any movement on this?

Been searching for a new note taking app, taking the free trial for a spin. Thought Quiver might be the solution for me, but, this issue may be a deal breaker.

Love the rest of the app so may still may move forward with it but really would love if this was fixed.

**Comment by ylian** [comment]
I am working on two big tasks: editor rewrite and app UI rewrite. This will be fixed with UI rewrite.
